v2/dns.go

```diff
@@ -0,0 +1,260 @@
+package v2
+
+import (
+	"context"
+	"time"
+
+	apiv2 "github.com/exoscale/egoscale/v2/api"
+	"github.com/exoscale/egoscale/v2/oapi"
+)
+
+// DnsDomain represents a DNS domain.
+type DnsDomain struct {
+	CreatedAt   *time.Time
+	ID          *string `req-for:"delete"`
+	UnicodeName *string `req-for:"create"`
+}
+
+// DnsDomainRecord represents a DNS record.
+type DnsDomainRecord struct {
+	Content   *string `req-for:"create"`
+	CreatedAt *time.Time
+	ID        *string `req-for:"delete,update"`
+	Name      *string `req-for:"create"`
+	Priority  *int64
+	Ttl       *int64
+	Type      *string `req-for:"create"`
+	UpdatedAt *time.Time
```

Comment:
It would be nice if we would avoid using pointer everywhere.
It might not be possible for all fields but for instance, `Name` cannot be an empty string, so if it is an empty string we can consider it as "not set".

This would move all the pointer checking/conversion from the user's code to here.
Downside it that this part of egoscale would be less consistent with other resources.

Reply:
Problem would be with `List*` EPs as they only guarantee `ID`, other fields may or may not be present. Lets revisit this later.